cpp/src/common/nvtx.cu

```diff
@@ -144,10 +146,10 @@ void PUSH_RANGE(const char *name) {
   eventAttrib.color = generateNextColor(name);
   eventAttrib.messageType = NVTX_MESSAGE_TYPE_ASCII;
   eventAttrib.message.ascii = name;
-  nvtxRangePushEx(&eventAttrib);
+  nvtxDomainRangePushEx(domain, &eventAttrib);
 }
 
-void POP_RANGE() { nvtxRangePop(); }
+void POP_RANGE() { nvtxDomainRangePop(domain); }
```

Comment:
CUDA synchronization should be required for both PUSH and POP operations. I just added a version of these functions that include synchronization. It would be interesting to discuss a bit more about it to see if we should use the synchronized version everywhere. It might still not produce perfect time measurements, especially when other CUDA streams are used in the background.